ls1intum/Artemis
# `Programming exercises`: Add gradle as project type for Java exercises on GitLab/Jenkins setup

### Checklist
#### General
#### Server
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite). (changes have no effect)

### Motivation and Context
Closes #4841.
In #4781, Gradle has been added as a project type for Java exercises on the Atlassian Setup. Within this PR, the Jenkins build plan creation is extended to also add the option for Gradle on the GitLab/Jenkins-Setup. Including support for
- Static Code Analysis and
- Sequential Test Runs.

### Description
The build plan creation for Java exercises is extended to distinct between Jenkinsfiles for Maven and Gradle projects.

### Steps for Testing
- 1 Instructor
- TS2 (GitLab/Jenkins-Setup)

1. Log in to Artemis
2. Navigate to Course Administration
3. Create Programming Exercises with the following configurations.
- Template Solution Repository Type: Plain or Gradle (does not matter)
- Test Repository Type: Gradle
- SCA: enabled; sequential test runs: disabled OR SCA: disabled, sequential test runs: enabled
4. Confirm that the initial result is created correctly.
5. Push some changes and verify that the results are built correctly.
6. Create a programming exercise with Maven and SCA enabled and confirm that everything works correctly.

### Review Progress
#### Code Review
#### Manual Tests
